ripe/atlas/tools/ipcache.py

```diff
@@ -0,0 +1,61 @@
+import time
```

Comment:
I'm not sure about this.  On the one hand, yes, we definitely need a clean
interface that abstracts away the RIPEStat call and caches the output, but I'm
not sure that I like the way this is currently setup: it's effectively a
function wrapped in an object, placed in a variable that then must be treated
like an object to call the method -- it feels convoluted.

Instead, an interface that worked like this would be much cleaner/easier to work
with:

```python
# ip.py

from .cache import cache

class Ip(object):

    def __init__(self, address):
        self.address = address
        self.asn = None
        self.holder = None
        ...
        self._populate()

    def self._populate(self)
        """
        Querys/caches all of the relevant data, populating self.asn, 
        self.holder, etc
        """

        # Pretty much all of your existing code here, but populating self,
        # rather than returning anything


# somewhere_else.py

from ip import Ip

ip = Ip.(address)
print(ip.asn)

ips = Ip.generate(address1, address2 address3)
print(ips[0].asn)
```

Does any of this make sense?  It's just that as it is, it's not really a cache
so much as a model with some built-in caching, and while I like this model, I'd
like it to conform to the so-called "pythonic" style.

Let me know what you think.